### ISSUE TITLE
[Bug]: Dropout with training_mode=true cannot be converted by the ONNX frontend (functional gap vs onnxruntime)

### ISSUE THREAD

**Issue by siyiweigeHEW** [openvinotoolkit/openvino#37767]
### OpenVINO Version

2026.1.0 (build 2026.1.0-21367, releases/2026/1)

### Operating System

Ubuntu — Ubuntu 20.04.6 LTS (Focal Fossa), Linux x86_64 (kernel 5.15.0-139-generic)

### Device used for inference

CPU

### Framework

ONNX

### Model used

Minimal ONNX model generated by the reproducer script.

### Issue description

OpenVINO's ONNX frontend cannot convert a valid `Dropout` node with `training_mode=true`, while onnxruntime executes the same model correctly. The inference path (`training_mode=false`) works as expected (the output is an identity copy and matches onnxruntime), so this is a functional gap versus the native runtime rather than an incorrect result:

```
Check '(!training_mode)' failed at src/frontends/onnx/frontend/src/op/dropout.cpp:20:
Training mode is not supported for Dropout op
```

The ONNX spec defines the training-mode computation (`output = data * mask / (1 - ratio)` with a random boolean `mask`), and onnxruntime implements it (see log below: `[2.0, 4.0, 6.0, 0.0]` with mask `[True, True, True, False]` for ratio 0.5). A similar training-mode limitation for `BatchNormalization` was recently lifted on the same frontend (it is now converted to a normalizing subgraph, PR #37170); `Dropout` training mode is the same class of gap.

OS: Linux x86_64 (kernel 5.15.0-139-generic)

### Step-by-step reproduction

```python
#!/usr/bin/env python3
import numpy as np
import onnx
import onnxruntime as ort
import openvino as ov
from onnx import TensorProto, helper

# ONNX Dropout (opset 12) with training_mode=true: a valid model that
# onnxruntime executes but OpenVINO cannot convert.
node = helper.make_node("Dropout", ["data", "ratio", "training_mode"], ["output", "mask"])

graph = helper.make_graph(
    [node],
    "dropout_training",
    [helper.make_tensor_value_info("data", TensorProto.FLOAT, [4])],
    [
        helper.make_tensor_value_info("output", TensorProto.FLOAT, [4]),
        helper.make_tensor_value_info("mask", TensorProto.BOOL, [4]),
    ],
    [
        helper.make_tensor("ratio", TensorProto.FLOAT, [], [0.5]),
        helper.make_tensor("training_mode", TensorProto.BOOL, [], [True]),
    ],
)

model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 12)])
model.ir_version = 9
onnx.checker.check_model(model)

data = np.array([1.0, 2.0, 3.0, 4.0], dtype=np.float32)
mb = model.SerializeToString()

ort_out, ort_mask = ort.InferenceSession(mb, providers=["CPUExecutionProvider"]).run(None, {"data": data})
print("onnxruntime output:", ort_out.tolist())
print("onnxruntime mask  :", ort_mask.tolist())

print("openvino attempt:")
try:
    ov.Core().read_model(mb)
    print("  converted OK")
except Exception as e:
    print("  FAILED TO CONVERT:")
    print("  " + str(e).replace("\n", "\n  "))
```

### Relevant log output

```shell
onnxruntime output: [2.0, 4.0, 6.0, 0.0]
onnxruntime mask  : [True, True, True, False]
openvino attempt:
  FAILED TO CONVERT:
  Exception from src/inference/src/cpp/core.cpp:106:
  Check 'false' failed at src/frontends/common_translators/src/unconverted_ops_report.cpp:141:
  FrontEnd API failed with OpConversionFailure:
  Model wasn't fully converted. Failed operations detailed log:
  -- Dropout-12 with a message:
  Check '(!training_mode)' failed at src/frontends/onnx/frontend/src/op/dropout.cpp:20:
  While validating ONNX node '<Node(Dropout): output, mask>':
  Training mode is not supported for Dropout op
  
  Summary:
  -- Conversion is failed for: Dropout-12
```

### Issue submission checklist

- [x] I'm reporting an issue. It's not a question.
- [x] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [x] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by YuChern-Intel** [comment]
Yes, this issue can be reproduced on my side.

<img width="669" height="221" alt="Image" src="https://github.com/user-attachments/assets/8e548577-a765-417c-8241-20c2c6871f51" />

I’ll escalate the case to relevant team for further investigation. We will provide an update soon.

**Comment by siyiweigeHEW** [comment]
> Yes, this issue can be reproduced on my side.
> 
> <img alt="Image" width="669" height="221" src="https://private-user-images.githubusercontent.com/107900329/645473257-8e548577-a765-417c-8241-20c2c6871f51.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg2MTE1NjIsIm5iZiI6MTc4ODYxMTI2MiwicGF0aCI6Ii8xMDc5MDAzMjkvNjQ1NDczMjU3LThlNTQ4NTc3LWE3NjUtNDE3Yy04MjQxLTIwYzJjNjg3MWY1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwOTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDkwNVQxMjI3NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTY3MmIzYzkzNTc0MjQyMWFjMDUyZDY5NTRhNmY3YmM3OTY1YTk3YTMxOTkwYzAzM2ViNzFhNmNhODJhODkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ._fXuLB-YaLDHEdtZvzimjDsaAoHAEmKZEvIbMfrhUo0">
> I’ll escalate the case to relevant team for further investigation. We will provide an update soon.
Thanks for confirming and for looking into this!

I’ll also try to investigate the issue and see if I can come up with a fix on my side. I’m not yet sure whether my approach is correct, but I’ll give it a try and share my findings once I have something.

**Comment by YuChern-Intel** [comment]
After investigation, it was confirmed that Dropout with training mode is not supported. Hence, the error message shown is actually expected.

**Comment by diego-villalobos** [comment]
Ref. 194273

**Comment by YuChern-Intel** [comment]
For your information, here is the [PR](https://github.com/openvinotoolkit/openvino/pull/38024) that will update the documentation and clarify that Dropout with training mode is not supported.

Once merged, the updated information will be reflected in the official OpenVINO documentation here:
https://docs.openvino.ai/2026/documentation/compatibility-and-support/supported-operations.html
